lib/logstash/outputs/loggly.rb

```diff
@@ -117,17 +123,50 @@ def send_event(url, message)
       http.verify_mode = OpenSSL::SSL::VERIFY_NONE
     end
 
-    # post message
     request = Net::HTTP::Post.new(url.path)
     request.body = message
-    response = http.request(request)
 
-    if response.is_a?(Net::HTTPSuccess)
-      @logger.info("Event send to Loggly OK!")
-    else
-      @logger.warn("HTTP error", :error => response.error!)
+    # Variable for count total retries
+    totalRetries = 0
+
+    #try posting once when can_retry is false
+    if @can_retry == false
+      @retry_count = 1
+    end
+
+    
+    while totalRetries < @retry_count
+    begin
+
+      response = http.request(request)	
+      case response.code
+	    when "200"					# HTTPSuccess :Code 2xx
```

Comment:
You might consider using constants here. I try to use constants to avoid having a typo ruin my day.

```ruby
HTTP_OK = "200"

...

when HTTP_OK
  ...
```